march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -317,33 +317,35 @@ void MarchHardwareInterface::write(const ros::Duration& elapsed_time)
   }
 }
 
-void MarchHardwareInterface::resetIMotionCubesUntilTheyWork()
+void MarchHardwareInterface::initiateIMC()
 {
-  bool encoderSetCorrectly = false;
-
-  while (!encoderSetCorrectly)
+  for (int i = 0; i < num_joints_; ++i)
   {
-    encoderSetCorrectly = true;
-    for (int i = 0; i < num_joints_; ++i)
+    march4cpp::Joint joint = marchRobot.getJoint(joint_names_[i]);
```

Comment:
It is probably nicer to loop using a foreach like so:
```c++
for (march4cpp::Joint& joint : marchRobot.jointList) {
...
}
```
And then even it would be nicer to make the `jointList` member private and create a `getJoints` method. And also add `using march4cpp::Joint;` at the top of the file, so that hideous namespace does not have to be defined every time. :slightly_smiling_face: